openedx/core/djangoapps/course_groups/migrations/0006_cohort_membership_data_migrate.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        # Note that we're using all() instead of filtering on CourseUserGroup.COHORT, due to South freezing the ORM.
+        # The overriden clean() method in CohortMembership will keep these invalid entries from saving.
+        for cohort_group in orm.CourseUserGroup.objects.all():
+            for user in cohort_group.users:
+                current_course_groups = orm.CourseUserGroup.objects.filter(
+                    course_id=cohort.course_id,
+                    users__id=user.id,
+                )
+                current_user_groups = user.course_groups.filter(
+                    course_id=cohort.course_id
+                )
+
+                unioned_set = set(current_course_groups).union(set(current_user_groups))
+
+                # Per product guidance, fix problem users by arbitrarily choosing a single membership to retain
+                cohort = unioned_set[0]
```

Comment:
I have not yet tested this, but I will this afternoon. `pop()` should do what I want here, correct?